### PR TITLE
Software development

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
Adds:
- [x]  exercise on good practices (from coderefinery in 1 hour crash course)

Creates the structure for including:
- [x]  introductory slides to the hands-on part on software development
- [x]  exercise on ci/cd (from Sanders cicd template)
- [x] exercise on measuring and reducing co2 emissions from your code using code carbon

### CODE REVIEW

**Comment by JaroCamphuijsen** [comment]
@raar1 could you provide a link to sanders ci/cd stuff?

**Comment by raar1** [comment]
> @raar1 could you provide a link to sanders ci/cd stuff?

So this is the link to the github action: <https://github.com/marketplace/actions/pytest-last-failed>

Recently he also suggested <https://github.com/tarpas/pytest-testmon> and potentially we can mention <https://testmon.org/blog/better-github-actions-caching/>

**Comment by JaroCamphuijsen** [comment]
Decided to reduce the scope of this PR and merge it into main so we can work on various chapters simultaneously. The following issues contain the remaining open todo's from this pr:

-  introductory slides to the hands-on part on software development #24 
- exercise on ci/cd (from Sanders cicd template) #25 
- exercise on measuring and reducing co2 emissions from your code using code carbon #26

**Comment by JaroCamphuijsen** [comment]
Decided to reduce the scope of this PR and merge it into main so we can work on various chapters simultaneously. The following issues contain the remaining open todo's from this pr:

-  introductory slides to the hands-on part on software development #24 
- exercise on ci/cd (from Sanders cicd template) #25 
- exercise on measuring and reducing co2 emissions from your code using code carbon #26

File: modules/software-development-handson/info.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ order: 0
 
 In this module you will learn to answer the following questions:
 
+* How do I set up or adapt my continuous integration and deployment system to minimize energy usage?
+* What are good triggers and intervals to run my various types of tests?
```
Comment (raar1) [inline modules/software-development-handson/info.md:12]: Nice, glad you updated this

**Review by raar1 — APPROVED** [review]
Thanks @JaroCamphuijsen this looks great. I like the structure, and what you added for the condensed code refinery course (best practices stuff).

**Pull request merged**